### DISCUSSION TITLE
How do I turn off diagnostics on startup?

### DISCUSSION THREAD

**Discussion by stackmystack** [LazyVim/LazyVim#2215] [Q&A]
I am new to neovim and lazy.vim and all of that.

I tried to check out the source code, and did a lot of customisations (I really like the code base because it's simple and readable even for someone who isn't fluent in Lua, but fluent in other languages) but I don't think I quite got to where I want to actually do it.

**Comment by dpetka2001** [comment]
Do you want to disable the diagnostics virtual text? Or the diagnostics as a whole?

**Reply by stackmystack** [reply]
Disable as if I pressed `<space>ud` => the diagnostics are detected, but the virtual text disabled on startup / by default.

**Reply by dpetka2001** [reply]
You could try the following for virtual text and signs
```lua
-- ~/.config/nvim/lua/plugins/lsp.lua
return {
  {
    "nvim-lspconfig",
    opts = {
      diagnostics = {
        virtual_text = false,
        signs = false,
      },
    },
  },
}
```
You could keep the signs in the status column if you'd like and only disable the virtual text. Up to you.

**Reply by stackmystack** [reply]
But then `<space>ud` would not work :/

I just want it `<space>ud`ed by default, so I could turn it on when I want to.

**Reply by stackmystack** [reply]
I think I found it:

```lua
-- ~/.config/nvim/lua/config/options.lua

require("lazyvim.util").toggle.diagnostics()

```

It will pop up a small notification at the beginning, but it's fine. I can hack my way from here :) thanks anyway!

**Reply by dpetka2001** [reply]
You could put instead `vim.diagnostic.disable()` to get rid of the initial message, but then that would mean when you press `<leader>ud` it would still toggle them off first and then the second time would toggle them on (because initially `enabled = true`) in the logic of `lazyvim.util.toggle.diagnostics`. Pick what you like best.

**Reply by aditya-an1l** [reply]
vim.diagnostic.disable() is deprecated. Use the following instead:

`vim.diagnostic.enable(false)`

**Reply by kickymaulana** [reply]
> You could try the following for virtual text and signs
> 
> ```lua
> -- ~/.config/nvim/lua/plugins/lsp.lua
> return {
>   {
>     "nvim-lspconfig",
>     opts = {
>       diagnostics = {
>         virtual_text = false,
>         signs = false,
>       },
>     },
>   },
> }
> ```
> 
> You could keep the signs in the status column if you'd like and only disable the virtual text. Up to you.

berhasil makasi banyak
